src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperAttributes.cs

```diff
@@ -0,0 +1,234 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    /// <summary>
+    /// Represents a collection of <see cref="TagHelperAttribute"/>s.
+    /// </summary>
+    public class TagHelperAttributes : ReadOnlyTagHelperAttributes<TagHelperAttribute>, IList<TagHelperAttribute>
+    {
+        /// <summary>
+        /// Instantiates a new instance of the <see cref="TagHelperAttributes"/> with an empty collection.
+        /// </summary>
+        public TagHelperAttributes()
+            : base()
+        {
+        }
+
+        /// <summary>
+        /// Instantiates a new instance of the <see cref="TagHelperAttributes"/> with the specified
+        /// <paramref name="attributes"/>.
+        /// </summary>
+        /// <param name="attributes">The collection to wrap.</param>
+        public TagHelperAttributes([NotNull] IEnumerable<TagHelperAttribute> attributes)
+            : base(attributes)
+        {
+        }
+
+        /// <inheritdoc />
+        /// <remarks><paramref name="value"/>'s <see cref="TagHelperAttribute.Name"/> cannot be <c>null</c>.</remarks>
+        public new TagHelperAttribute this[int index]
```

Comment:
if `new` can be avoided, do it. (I can imagine a couple of reasons why `new` would be necessary here and in the next indexer but want to be sure.)